numpyro/distributions/distribution.py

```diff
@@ -558,9 +558,10 @@ def reshape_sample(x):
             event_shape = jnp.shape(x)[batch_ndims:]
             return x.reshape(sample_shape + self.batch_shape + event_shape)
 
+        # TODO: See why we need intermediates here
```

Comment:
we still need to double check this...

Reply:
I think we can add `intermediates` arguments to log_prob method to utilize this information. This is useful for Expanded(TransformDistribution).

Reply:
ok i'll leave this to you 👍